win32/tests/threadpool_win32_int/threadpool_win32_int.c

```diff
@@ -998,6 +998,139 @@ static DWORD WINAPI chaos_thread_with_timers_func(LPVOID lpThreadParameter)
+//                {
+//                    uint32_t timer_start_delay = TIMER_START_DELAY_MIN + rand() * (TIMER_START_DELAY_MAX - TIMER_START_DELAY_MIN) / (RAND_MAX + 1);
+//                    uint32_t timer_period = TIMER_PERIOD_MIN + rand() * (TIMER_PERIOD_MAX - TIMER_PERIOD_MIN) / (RAND_MAX + 1);
+//                    if (threadpool_timer_start(chaos_test_data->threadpool, timer_start_delay, timer_period, work_function, (void*)&chaos_test_data->executed_timer_functions, &chaos_test_data->timers[which_timer_slot].timer) == 0)
+//                    {
+//                        InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_STARTED);
+//                    }
+//                    else
+//                    {
+//                        InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_NONE);
+//                    }
+//                }
+//            }
+//            (void)InterlockedDecrement(&chaos_test_data->timers_starting);
+//            WakeByAddressSingle((PVOID)&chaos_test_data->timers_starting);
+//        }
+//        break;
+//        case 4:
+//            // Stop a timer
+//        {
+//            // Synchronize with close
+//            (void)InterlockedIncrement(&chaos_test_data->timers_starting);
+//            if (InterlockedAdd(&chaos_test_data->can_start_timers, 0) != 0)
+//            {
+//                int which_timer_slot = rand() * MAX_TIMER_COUNT / (RAND_MAX + 1);
+//                if (InterlockedCompareExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_STOPPING, TIMER_STATE_STARTED) == TIMER_STATE_STARTED)
+//                {
+//                    threadpool_timer_destroy(chaos_test_data->timers[which_timer_slot].timer);
+//                    //chaos_test_data->timers[which_timer_slot].timer = NULL;
+//                    InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_NONE);
+//                }
+//            }
+//            (void)InterlockedDecrement(&chaos_test_data->timers_starting);
+//            WakeByAddressSingle((PVOID)&chaos_test_data->timers_starting);
+//        }
+//        break;
+//        case 5:
+//            // Cancel a timer
+//        {
+//            // Synchronize with close
+//            (void)InterlockedIncrement(&chaos_test_data->timers_starting);
+//            if (InterlockedAdd(&chaos_test_data->can_start_timers, 0) != 0)
+//            {
+//                int which_timer_slot = rand() * MAX_TIMER_COUNT / (RAND_MAX + 1);
+//                if (InterlockedAdd(&chaos_test_data->timers[which_timer_slot].state, 0) == TIMER_STATE_STARTED)
+//                {
+//                    threadpool_timer_cancel(chaos_test_data->timers[which_timer_slot].timer);
+//                    //(void)InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_STARTED);
```

Comment:
>//(void)InterlockedExchange(&chaos_test_data->timers[which_timer_slot].state, TIMER_STATE_STARTED); [](http://example.com/codeflow?start=22&length=99)

This seems to be commented out, should it?

Reply:
removed